pkg/storage/azure/azure.go

```diff
@@ -457,9 +452,9 @@ func (d *driver) CreateStorage(cr *imageregistryv1.Config) error {
 					return err
 				}
 				d.Config.AccountName = accountName
+				cr.Spec.Storage.Azure = d.Config.DeepCopy()
```

Comment:
I think the best way for us to detect if we created the storage account is to not overwrite `spec` when we create storage. This should probably be done in a separate PR since s3 overwrites `spec` when we create storage.

Reply:
This is just writing AccountName back to Spec. d.Config is a copy of cr.Spec.

Reply:
And if you are saying we shouldn't touch Spec at all, it will be backward incompatible change. Can we do it?